lib/n1_loader/core/loader.rb

```diff
@@ -23,7 +23,9 @@ def argument(name, **opts)
 
         @arguments ||= []
 
-        define_method(name) { args[name] ||= opts[:default]&.call }
+        define_method(name) do
+          args.fetch(name) { opts[:default]&.call }
+        end
```

Comment:
I don't recall if that's important but the previous version also was assigning the value to `args[name]`. 

Can we keep that feature too in case it's important? 
It can be fixed as simple as this:
```
args.fetch(name) { args[name] = opts[:default]&.call }
```

Reply:
:+1: 

My guess is the previous version was a performance optimization to the following alternative:
```ruby
define_method(name) { args[name] || opts[:default]&.call }
```
I did not think about it initially.